CRM/Contact/BAO/Contact/Location.php

```diff
@@ -40,16 +40,22 @@ class CRM_Contact_BAO_Contact_Location {
    *
    * @param bool $isPrimary
    * @param int $locationTypeID
+   * @param bool $isBilling
    *
    * @return array
    *   Array of display_name, email, location type and location id if found, or (null,null,null, null)
    */
-  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL) {
+  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL, $isBilling = FALSE) {
     $primaryClause = NULL;
     if ($isPrimary) {
       $primaryClause = " AND civicrm_email.is_primary = 1";
     }
 
+    // CRM-20362 retrieving Billing email address.
+    if ($isBilling) {
+      $primaryClause = " AND civicrm_email.is_billing = 1";
```

Comment:
> One thing would be is what if you have 2 billing email addresses? as that is something i know the GUI would allow

@seamuslee001 only the first defined billing email is set as "is_billing". If this first billing email is deleted, then the second billing email would be set as "is_billing"